flowmachine/flowmachine/features/subscriber/distance_counterparts.py

```diff
@@ -0,0 +1,151 @@
+        Either msisdn, or imei, the column that identifies the subscriber.
+    subscriber_subset : str, list, flowmachine.core.Query, flowmachine.core.Table, default None
+        If provided, string or list of string which are msisdn or imeis to limit
+        results to; or, a query or table which has a column with a name matching
+        subscriber_identifier (typically, msisdn), to limit results to.
+    statistic :  {'count', 'sum', 'avg', 'max', 'min', 'median', 'mode', 'stddev', 'variance'}, default 'avg'
+        Defaults to sum, aggregation statistic over the durations.
+
+
+    Examples
+    --------
+
+    >>> s = DistanceCounterparts("2016-01-01", "2016-01-07", statistic="avg")
+    >>> s.get_dataframe()
+
+              subscriber    distance_avg
+        038OVABN11Ak4W5P      272.167815
+        09NrjaNNvDanD8pk      241.290233
+        0ayZGYEQrqYlKw6g      218.161568
+        0DB8zw67E9mZAPK2      228.235324
+        0Gl95NRLjW2aw8pW      189.008980
+                     ...             ...
+
+    """
+
+    def __init__(
+        self,
+        start,
+        stop,
+        statistic="avg",
+        *,
+        hours="all",
+        tables="all",
+        direction="both",
+        exclude_self_calls=True,
+        subscriber_subset=None,
+    ):
+        self.tables = tables
+        self.start = start
+        self.stop = stop
+        self.hours = hours
+        self.direction = direction
+        self.exclude_self_calls = exclude_self_calls
+
+        self.statistic = statistic.lower()
+        if self.statistic not in valid_stats:
+            raise ValueError(
+                "{} is not a valid statistic. Use one of {}".format(
+                    self.statistic, valid_stats
+                )
+            )
+
+        column_list = ["msisdn", "msisdn_counterpart", "id", "location_id", "outgoing"]
+        self.tables = parse_tables_ensuring_columns(
+            self.connection, tables, column_list
+        )
+
+        # EventsTablesUnion will only subset on the subscriber identifier,
+        # which means that we need to query for a unioned table twice.
+        self.unioned_from_query = EventsTablesUnion(
+            self.start,
+            self.stop,
+            columns=column_list,
+            tables=self.tables,
+            subscriber_identifier="msisdn",
+            hours=hours,
+            subscriber_subset=subscriber_subset,
+        ).get_query()
+
+        self.unioned_to_query = EventsTablesUnion(
+            self.start,
+            self.stop,
+            columns=column_list,
+            tables=self.tables,
+            subscriber_identifier="msisdn_counterpart",
+            hours=hours,
+            subscriber_subset=subscriber_subset,
+        ).get_query()
+
+        self.distance_matrix = DistanceMatrix()
+
+        super().__init__()
+
+    def _make_query(self):
+
+        filters = []
+        if self.direction != "both":
```

Comment:
A similar comment applies here as in the other PR: would be good to check that direction actually has a valid value. However, this really should be solved more properly so I've opened #357 for this.